src/lib/containers/entity_finder/details/view/DetailsViewRow.tsx

```diff
@@ -81,22 +87,31 @@ export const DetailsViewRow: React.FunctionComponent<DetailsViewRowProps> = ({
   )
 
   useEffect(() => {
-    if (inView && isSelected && versions === undefined) {
+    if (isError && error) {
+      handleError(toError(error))
+    }
+  }, [isError, error, handleError])
+
+  useEffect(() => {
+    if (isSelected && versions === undefined) {
       SynapseClient.getEntityVersions(sessionToken, entityHeader.id).then(
         response => {
           setVersions(response.results)
         },
+        error => {
+          handleError(error)
+        },
       )
     }
-  }, [inView, isSelected, versions, sessionToken, entityHeader.id])
+  }, [isSelected, versions, sessionToken, entityHeader.id])
 
   return (
     <tr
       ref={ref}
-      className={`EntityFinderDetailsView__Row${
-        isSelected ? ' EntityFinderDetailsView__Row__Selected' : ''
-      }${isDisabled ? ' EntityFinderDetailsView__Row__Disabled' : ''}`}
-      style={isHidden ? { display: 'none' } : {}}
+      aria-selected={isSelected}
+      aria-disabled={isDisabled}
+      aria-hidden={isHidden}
+      className="EntityFinderDetailsView__Row"
```

Comment:
React Testing Library intentionally makes it difficult to inspect the DOM (e.g. validate CSS classes) and instead requires testing the accessibility tree. Thus we use ARIA attributes instead of custom classes. I think it ends up being neater as well.

Reply:
I agree!  This is something we should work towards in the rest of the codebase.